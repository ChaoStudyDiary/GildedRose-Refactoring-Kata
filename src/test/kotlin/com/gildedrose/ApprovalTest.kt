package com.gildedrose

import org.approvaltests.Approvals
import org.approvaltests.combinations.CombinationApprovals
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class ApprovalTest {

    @Test
    fun testApproval() {
        val items = listOf(Item("foo", 0, 10))
        val app = GildedRose(items)
        app.updateQuality()
        Approvals.verifyAll("", app.items)
    }

    @Test
    fun updateQuality() {
        CombinationApprovals.verifyAllCombinations(
            this::callUpdateQuality,
            arrayOf("a common item", "Sulfuras, Hand of Ragnaros", "Backstage passes to a TAFKAL80ETC concert","Aged Brie"),
            arrayOf(0, -1, 20),
            arrayOf(0, 1, 80, 49, 50)
        )
    }


    private fun callUpdateQuality(name: String, sellIn: Int, quality: Int): String {
        val items = arrayListOf(Item(name, sellIn, quality))
        val gildedRose = GildedRose(items)
        gildedRose.updateQuality()
        return gildedRose.items[0].toString()
    }
}