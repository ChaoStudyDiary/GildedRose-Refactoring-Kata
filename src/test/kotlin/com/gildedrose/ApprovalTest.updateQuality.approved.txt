[a common item, 0, 0] => a common item, -1, 0 
[a common item, 0, 1] => a common item, -1, 0 
[a common item, 0, 80] => a common item, -1, 78 
[a common item, 0, 49] => a common item, -1, 47 
[a common item, 0, 50] => a common item, -1, 48 
[a common item, -1, 0] => a common item, -2, 0 
[a common item, -1, 1] => a common item, -2, 0 
[a common item, -1, 80] => a common item, -2, 78 
[a common item, -1, 49] => a common item, -2, 47 
[a common item, -1, 50] => a common item, -2, 48 
[a common item, 20, 0] => a common item, 19, 0 
[a common item, 20, 1] => a common item, 19, 0 
[a common item, 20, 80] => a common item, 19, 79 
[a common item, 20, 49] => a common item, 19, 48 
[a common item, 20, 50] => a common item, 19, 49 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 80] => Sulfuras, Hand of Ragnaros, 0, 80 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 80] => Sulfuras, Hand of Ragnaros, -1, 80 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, 20, 0] => Sulfuras, Hand of Ragnaros, 20, 0 
[Sulfuras, Hand of Ragnaros, 20, 1] => Sulfuras, Hand of Ragnaros, 20, 1 
[Sulfuras, Hand of Ragnaros, 20, 80] => Sulfuras, Hand of Ragnaros, 20, 80 
[Sulfuras, Hand of Ragnaros, 20, 49] => Sulfuras, Hand of Ragnaros, 20, 49 
[Sulfuras, Hand of Ragnaros, 20, 50] => Sulfuras, Hand of Ragnaros, 20, 50 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 80] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 80] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 20, 0] => Backstage passes to a TAFKAL80ETC concert, 19, 1 
[Backstage passes to a TAFKAL80ETC concert, 20, 1] => Backstage passes to a TAFKAL80ETC concert, 19, 2 
[Backstage passes to a TAFKAL80ETC concert, 20, 80] => Backstage passes to a TAFKAL80ETC concert, 19, 80 
[Backstage passes to a TAFKAL80ETC concert, 20, 49] => Backstage passes to a TAFKAL80ETC concert, 19, 50 
[Backstage passes to a TAFKAL80ETC concert, 20, 50] => Backstage passes to a TAFKAL80ETC concert, 19, 50 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 80] => Aged Brie, -1, 80 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 80] => Aged Brie, -2, 80 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 20, 0] => Aged Brie, 19, 1 
[Aged Brie, 20, 1] => Aged Brie, 19, 2 
[Aged Brie, 20, 80] => Aged Brie, 19, 80 
[Aged Brie, 20, 49] => Aged Brie, 19, 50 
[Aged Brie, 20, 50] => Aged Brie, 19, 50 
